controllers/flp/flp_controller.go

```diff
@@ -269,8 +269,9 @@ func reconcileDataAccessRoles(ctx context.Context, r *reconcilers.Common, b *bui
 			}
 		}
 	}
-	// Install netobserv-metrics-reader role
-	return r.ReconcileClusterRole(ctx, &resources.PromReaderCR)
+	// Install netobserv-metrics-reader role; copy to avoid any undesired mutation
+	cr := resources.PromReaderCR
```

Comment:
do u need deepCopy here ?

Reply:
it's not strictly needed today but it's right that it would be safer in case we add more mutations later...
or alternatively we can have `PromReaderCR` being a function, rather than a global var

Reply:
@msherif1234 changed to use a function